### PR TITLE
add ddl callback interface

### PR DESCRIPTION
**Pull request by siddontang**
we can customize the callback for better test.

@ngaut  @qiuyesuifeng

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -165,6 +156,20 @@ func (do *Domain) loadSchemaInLoop(lease time.Duration) {
...
 }
 
+type ddlCallback struct {
+	ddl.BaseCallback
+	do *Domain
+}
+
+func (c *ddlCallback) OnChanged(err error) error {
+	if err != nil {
+		return err
```
Comment (qiuyesuifeng) [inline domain/domain.go]: errors.Trace(err)

Comment (siddontang) [inline domain/domain.go]: no need here, the error will be returned directly, and outer will use trace.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**